### ISSUE TITLE
Possible issue with DWORD -> INT64 conversion

### ISSUE THREAD

**Issue by cpyrgas** [X-Sharp/XSharpPublic#357]
Following code in 2.3b prints:

DWORD 4294967295
4294967295

in previous builds it would return

INT64 -1
-1

Not sure if it should be considered a bug, added for your consideration Robert.

```
FUNCTION Start() AS VOID
	LOCAL dw AS DWORD
	LOCAL i64 AS INT64
	dw := 1
	Test(-dw)

	i64 := -dw
	? i64
RETURN 

FUNCTION Test(dw AS DWORD) AS VOID
? "DWORD", dw
FUNCTION Test(i64 AS INT64) AS VOID
? "INT64", i64
```

**Comment by cpyrgas** [comment]
This issue has been brought already a few times, so I think we should provide a quick fix. Here's a different aspect of the same issue:

With overflow checking enabled, the following code compiles and runs in VO with no problems, In X#, it currently throws an exception at runtime:

Unhandled Exception: System.OverflowException: Arithmetic operation resulted in an overflow.
   at xRuntime.Exe.Functions.Start() in test.prg:line 5


```
// ovf+
FUNCTION Start() AS VOID
LOCAL u AS USUAL
LOCAL d AS DWORD
d := 1
u := - d
? u
```

**Comment by RobertvanderHulst** [comment]
Fixed in 91fb3ac17e63f220d1ac0594f67888461f3dd589
